base/strings/types.jl

```diff
@@ -219,3 +219,44 @@ julia> "Test "^3
 
 pointer(x::SubString{String}) = pointer(x.string) + x.offset
 pointer(x::SubString{String}, i::Integer) = pointer(x.string) + x.offset + (i-1)
+
+
+"""
+    SecureString(string::AbstractString)
+
+A string where the contents will be securely wiped when the last reference to the secure
+string has been removed. If `string` is of type `String` then the memory of the original
+parameter will also be securely wiped.
+
+# Examples
+```jldoctest
+julia> str = "abc"
+"abc"
+
+julia> SecureString(str)
+"abc"
+
+julia> gc()  # Cause the SecureString above to be garbage collected
+
+julia> str
+"\0\0\0"
+```
+"""
+mutable struct SecureString <: AbstractString
+    string::String
+
+    function SecureString(str::AbstractString)
+        s = new(str)
+        finalizer(securezero!, s)
```

Comment:
I don't understand a design based on finalizers, since finalizers may take a long time to call.   Don't you need to zero the data as soon as it falls out of scope?

Reply:
Best practise would be to securely zero the data as soon it is no longer in use. The finalizer is used mostly as a fail-safe to ensure that at the very least the data is zeroed when the the instance is garbage collected.

Reply:
I'll make sure to mention this in the docstring for `SecureString`